StonehearthEditor/Recipes/RecipesView.cs

```diff
@@ -541,26 +586,38 @@ private void searchBox_Filter(object sender, EventArgs e)
 
         private void recipesGridView_KeyDown(object sender, KeyEventArgs e)
         {
-            // Delete cell on del
             if (e.KeyCode == Keys.Delete)
             {
+                // Delete cell on del
                 DeleteCurrentCell();
             }
-
-            // Save on ctrl+s
             else if (e.Control && e.KeyCode == Keys.S)
             {
+                // Save on ctrl+s
                 SaveModifiedFiles();
             }
-
-            // Paste multiple cells on ctrl+v
+            else if (e.Control && e.KeyCode == Keys.Z)
+            {
+                // Undo on ctrl+z
+                ApplyCellChanges(mUndoStack, mRedoStack, changes => changes.Undo());
+            }
+            else if (e.Control && e.KeyCode == Keys.Y)
```

Comment:
Can we also have Ctrl+Shift+Z for Redo? Most programs use both and Ctrl+Shift+Z is easier to type.

Reply:
sure, it will have both